wandb/sdk/wandb_setup.py

```diff
@@ -161,6 +163,12 @@ def _settings_setup(
         self._logger.info(f"Configure stats pid to {pid}")
         s.x_stats_pid = pid
 
+        if settings and settings.settings_system:
+            s.settings_system = os.path.join("~", ".config", "wandb", "settings")
```

Comment:
This should be `settings.settings_system`, shouldn't it? For the default value, you probably just want an `else` case at the end of this if-else chain.

Use `pathlib.Path("~", ".config", "wandb", "settings").expanduser()` as the modern alternative.

Reply:
Right, what I meant was to set `s.settings_system` to `settings.settings_system` instead of setting it to `os.path.join(...)`. The updated PR looks good!